edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/CloudProxy.cs

```diff
@@ -152,12 +153,18 @@ public async Task SendMessageAsync(IMessage inputMessage)
                 string outputRoute = inputMessage.GetOutput();
                 using (Metrics.TimeMessageSend(this.clientId, outputRoute))
                 {
+                    var tokenSource = new CancellationTokenSource(this.sdkWaitTime);
                     Metrics.MessageProcessingLatency(this.clientId, inputMessage);
-                    await this.client.SendEventAsync(message);
+                    await this.client.SendEventAsync(message, tokenSource.Token).TimeoutAfterSDKHang(TimeSpan.FromSeconds(50));
                     Events.SendMessage(this);
```

Comment:
The hardcoded 50 is not so nice. Maybe it could be a second readonly value besides the sdkWaitTime or derive it from sdkWaitTime  (e.g.  sdkWaitTime + 5)

Reply:
+1